src/generic/grideditors.cpp

```diff
@@ -1266,14 +1245,26 @@ void wxGridCellBoolEditor::SetSize(const wxRect& r)
 
     // position it in the centre of the rectangle (TODO: support alignment?)
 
+    wxRect cellRect = r;
+
 #if defined(__WXGTK__) || defined (__WXMOTIF__)
+#if defined(__WXGTK3__)
+    cellRect.x += 6;
+    cellRect.width -= 12;
+#else
```

Comment:
This seems to be a confusing way to write `#elif defined(__WXMOTIF__)`.

In any case, let's just drop the Motif tests, nobody cares about it anyhow and a wrong offset in a grid editor is the least of its problems.